addon/components/listing.hbs

```diff
@@ -1,32 +1,98 @@
-<div class="au-o-grid">
-  <div class="au-o-grid__item au-u-4-5">
-    {{#each this.subForms as |subForm index|}}
-      <SubForm
-        @subForm={{subForm}}
-        @formStore={{@formStore}}
-        @graphs={{@graphs}}
-        @sourceNode={{subForm.sourceNode}}
-        @forceShowErrors={{@forceShowErrors}}
-        @cacheConditionals={{@cacheConditionals}}
-        @last={{eq index (sub this.subForms.length 1)}}
-        @show={{@show}}
-        @level={{@level}}
-        @onRemoveEntry={{this.removeEntry}}
-        @canRemove={{@listing.canRemove}}
-      />
-    {{/each}}
-  </div>
-  {{#if this.canAdd}}
-    <div class="au-o-grid__item au-u-1-5">
+{{#if @listing.isTable}}
+  <div>
+    <AuTable class="listing-table">
+      <:title>
+        {{! TODO: read this from the graph directly }}
+        {{this.subForms.firstObject.itemLabel}}
+      </:title>
+      <:header>
+        {{!TODO this can probably be done without the subforms setup by simply retrieving the labels from the graph in some way?}}
+        {{#let this.subForms.firstObject as |subForm|}}
+          {{#if subForm}}
+            <SubForm
+              @subForm={{subForm}}
+              @formStore={{@formStore}}
+              @graphs={{@graphs}}
+              @sourceNode={{subForm.sourceNode}}
+              @forceShowErrors={{@forceShowErrors}}
+              @cacheConditionals={{@cacheConditionals}}
+              @last={{false}}
+              @show={{@show}}
+              @level={{@level}}
+              @onRemoveEntry={{this.removeEntry}}
+              @canRemove={{@listing.canRemove}}
+              @isRow={{true}}
+              @isTableHead={{true}}
+            />
+          {{/if}}
+        {{/let}}
+      </:header>
+      <:body>
+        {{! TODO: Show something if the table is empty }}
+        {{#each this.subForms as |subForm index|}}
```

Comment:
I've reused the subForm system of regular listings since that keeps the form.ttl the same. Would it be better to go with a different setup specifically made for tables instead?

Reply:
yeah the more i look at it the more it feels like this no?